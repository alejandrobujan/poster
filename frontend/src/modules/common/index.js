export {default as Pager} from "./components/Pager";
export {default as Errors} from "./components/Errors";
export {default as BackLink} from "./components/BackLink";
export {default as UserCard} from "./components/UserCard";
export {default as PostLink} from "./components/PostLink";
export {default as ImagesCard} from "./components/ImagesCard";