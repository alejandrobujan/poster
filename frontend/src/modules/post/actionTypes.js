export const CLEAR_POST_SEARCH = "poster/post/clearPostSearch";
export const FIND_POSTS_COMPLETED = "poster/post/findPostsCompleted";