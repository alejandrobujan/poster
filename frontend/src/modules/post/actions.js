import backend from '../../backend';
import * as actionTypes from './actionTypes';
import * as selectors from './selectors';

const findAllCategoriesCompleted = categories => ({
	type: actionTypes.FIND_ALL_CATEGORIES_COMPLETED,
	categories
});

export const findAllCategories = () => (dispatch, getState) => {

	const categories = selectors.getCategories(getState());

	if (!categories) {

		backend.postService.findAllCategories(
			categories => dispatch(findAllCategoriesCompleted(categories))
		);

	}

}

export const createPost = (title, description, url, price, categoryId, images, type, properties,
	onSuccess, onErrors) => dispatch =>
		backend.postService.createPost(title, description, url, price, categoryId, images, type, properties, onSuccess, onErrors);

const findPostsCompleted = postSearch => ({
	type: actionTypes.FIND_POSTS_COMPLETED,
	postSearch
});


const clearPostSearch = () => ({
	type: actionTypes.CLEAR_POST_SEARCH
});


export const findPosts = criteria => dispatch => {

	dispatch(clearPostSearch());
	backend.postService.findPosts(criteria, result =>
		dispatch(findPostsCompleted({ criteria, result }))
	);

};

export const previousFindPostsResultPage = criteria =>
	findPosts({ page: criteria.page - 1 });

export const nextFindPostsResultPage = criteria =>
	findPosts({ page: criteria.page + 1 });

export const findPostById = id => dispatch => {
	backend.postService.findPostById(id,
		post => dispatch(findPostByIdCompleted(post)));
}

const findPostByIdCompleted = post => ({
	type: actionTypes.FIND_POST_BY_ID_COMPLETED,
	post
});

export const clearPost = () => ({
	type: actionTypes.CLEAR_POST
}); 	

export const deletePost = (id, onSuccess, onErrors) => dispatch => {
	backend.postService.deletePost(id, onSuccess, onErrors);
};


export const updatePostCompleted = post => ({
    type: actionTypes.UPDATE_POST_COMPLETED,
    post
})
    
export const updatePost = (post, onSuccess, onErrors) => dispatch =>
	backend.postService.updatePost(post,
		post => {
			dispatch(updatePostCompleted(post));
			onSuccess();
		}, 
		onErrors);

const markPostAsExpiredCompleted = ({expired}) => ({
	type: actionTypes.MARK_POST_AS_EXPIRED_COMPLETED,
	expired
});

export const markPostAsExpired = (id, expired, onErrors) => dispatch => {
	backend.postService.maskPostAsExpired(id, expired, e => dispatch(markPostAsExpiredCompleted(e)), onErrors);
};	

const ratePostCompleted = ratingCount => ({
	type: actionTypes.RATE_POST_COMPLETED,
	ratingCount
});

export const ratePostPositive = (id, onErrors) => dispatch => {
	backend.ratingService.ratePostPositive(id, ratingCount => dispatch(ratePostCompleted(ratingCount)), onErrors);
};


export const ratePostNegative = (id, onErrors) => dispatch => {
	backend.ratingService.ratePostNegative(id, ratingCount => dispatch(ratePostCompleted(ratingCount)), onErrors);
};
