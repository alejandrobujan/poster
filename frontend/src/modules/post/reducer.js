import {combineReducers} from 'redux';

import * as actionTypes from './actionTypes';

const initialState = {
    post: null
};

const post = (state = initialState.post, action) => {
    switch(action.type) {

        case actionTypes.FIND_POST_BY_ID_COMPLETED:
            return action.post;

        case actionTypes.CLEAR_POST:
            return initialState.post;    
            
        case actionTypes.MARK_POST_AS_EXPIRED_COMPLETED:
			return {...state, expired: action.expired };
			
		case actionTypes.RATE_POST_COMPLETED:
			return { ...state, positiveRatings: action.ratingCount.positiveCount, negativeRatings: action.ratingCount.negativeCount };	
		
		case actionTypes.UPDATE_POST_COMPLETED:
			return action.post;
			
		case actionTypes.MARK_POST_AS_VALID_COMPLETED:
			return {...state, validationDate: action.validationDate };

        default:
            return state;
    }
}



const reducer = combineReducers({
    post
});

export default reducer;