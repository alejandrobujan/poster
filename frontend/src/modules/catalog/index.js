import * as actions from './actions';
import * as actionTypes from './actionTypes';
import reducer from './reducer';
import * as selectors from './selectors';

export {default as SearchBar} from "./components/SearchBar";
export {default as CategorySelector} from './components/CategorySelector';
export {default as Post} from './components/Post';
export {default as Posts} from './components/Posts';
export {default as Feed} from './components/Feed';

// eslint-disable-next-line
export default {actions, actionTypes, reducer, selectors};


