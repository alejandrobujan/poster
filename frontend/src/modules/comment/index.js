import * as actions from './actions';
import reducer from './reducer';
import * as selectors from './selectors';

export {default as Comments} from './components/Comments';
export {default as Comment} from './components/Comment';

// eslint-disable-next-line
export default  {actions, reducer, selectors};