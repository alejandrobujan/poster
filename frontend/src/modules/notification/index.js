import * as actions from './actions';
import * as actionTypes from './actionTypes';
import reducer from './reducer';
import * as selectors from './selectors';

export {default as Notifications} from "./components/Notifications";
export {default as Notification} from "./components/Notification";

// eslint-disable-next-line
export default {actions, actionTypes, reducer, selectors};


