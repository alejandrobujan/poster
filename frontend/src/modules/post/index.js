import * as actions from './actions';
import reducer from './reducer';
import * as selectors from './selectors';

export {default as CreatePost} from './components/CreatePost';
export {default as CategorySelector} from './components/CategorySelector';
export {default as Post} from './components/Post';
export {default as Posts} from './components/Posts';
export {default as Feed} from './components/Feed';
export {default as PostDetails} from './components/PostDetails';


// eslint-disable-next-line
export default  {actions, reducer, selectors};