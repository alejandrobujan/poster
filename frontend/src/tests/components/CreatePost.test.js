import React from "react";
import { MemoryRouter } from "react-router-dom";
import {Provider} from 'react-redux';
import renderer from "react-test-renderer";
import store from "../../store";
import {CreatePost} from "../../modules/post";

describe("Logout", () => {

    it("renders correctly", () => {
		const tree = renderer
        .create(
				<Provider store={store}>
					<MemoryRouter>
						<CreatePost/>
					</MemoryRouter>
				</Provider>
			)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});

});