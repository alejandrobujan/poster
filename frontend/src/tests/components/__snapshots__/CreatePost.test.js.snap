// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreatePost renders correctly 1`] = `
<div
  className="container"
>
  <div
    className="card bg-light border-dark m-4"
  >
    <h5
      className="card-header text-center"
    >
      Create post
    </h5>
    <div
      className="card-body"
    >
      <nav
        aria-label="page navigation"
      >
        <ul
          className="pagination justify-content-center"
        >
          <li
            className="page-item disabled"
          >
            <button
              className="page-link"
              onClick={[Function]}
            >
              Offer
            </button>
          </li>
          <li
            className="page-item "
          >
            <button
              className="page-link"
              onClick={[Function]}
            >
              Coupon
            </button>
          </li>
        </ul>
      </nav>
      <form
        className="needs-validation"
        noValidate={true}
        onSubmit={[Function]}
      >
        <div
          className="form-group row"
        >
          <label
            className="col-md-3 col-form-label"
            htmlFor="title"
          >
            Title (*)
          </label>
          <div
            className="col-md-9"
          >
            <input
              autoFocus={true}
              className="form-control"
              id="title"
              maxLength={16}
              minLength={1}
              onChange={[Function]}
              required={true}
              type="text"
              value=""
            />
            <div
              className="invalid-feedback"
            >
              The title size must be between 1 and 60
            </div>
          </div>
        </div>
        <div
          className="form-group row"
        >
          <label
            className="col-md-3 col-form-label"
            htmlFor="description"
          >
            Description (*)
          </label>
          <div
            className="col-md-9"
          >
            <textarea
              className="form-control"
              id="description"
              maxLength={256}
              minLength={1}
              onChange={[Function]}
              required={true}
              value=""
            />
            <div
              className="invalid-feedback"
            >
              The description size must be between 1 and 256
            </div>
          </div>
        </div>
        <div
          className="form-group row"
        >
          <label
            className="col-md-3 col-form-label"
            htmlFor="url"
          >
            Url
          </label>
          <div
            className="col-md-9"
          >
            <input
              className="form-control"
              id="url"
              maxLength={2048}
              minLength={0}
              onChange={[Function]}
              type="text"
              value=""
            />
            <div
              className="invalid-feedback"
            >
              The url size must be between 0 and 2048
            </div>
          </div>
        </div>
        <div
          className="form-group row"
        >
          <label
            className="col-md-3 col-form-label"
            htmlFor="price"
          >
            Price (*)
          </label>
          <div
            className="col-md-3"
          >
            <div
              className="input-group"
            >
              <input
                className="form-control"
                id="price"
                max="9999999.99"
                min="0.00"
                onChange={[Function]}
                required={true}
                step="0.01"
                type="number"
                value={0}
              />
              <span
                className="input-group-text"
              >
                â‚¬
              </span>
            </div>
            <div
              className="invalid-feedback"
            >
              Price must be between 0 and 9999999.99
            </div>
          </div>
          <label
            className="col-md-3 col-form-label"
            htmlFor="categoryId"
          >
            Category (*)
          </label>
          <div
            className="col-md-3"
          >
            <select
              className="form-control"
              id="categoryId"
              onChange={[Function]}
              value=""
            >
              <option
                style={
                  Object {
                    "color": "gray",
                    "fontStyle": "italic",
                  }
                }
                value=""
              >
                Select
              </option>
            </select>
          </div>
        </div>
        <div
          className="form-group row"
        >
          <label
            className="col-md-3 col-form-label"
            htmlFor="images"
          >
            Images
          </label>
          <div
            className="col-md-9"
          >
            <input
              accept="image/*"
              id="images"
              multiple={true}
              onChange={[Function]}
              type="file"
            />
            <button
              className="btn btn-outline-danger"
              onClick={[Function]}
              style={
                Object {
                  "display": "none",
                }
              }
              type="button"
            >
              <svg
                className="bi bi-trash"
                fill="currentColor"
                height="16"
                viewBox="0 0 16 16"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M5.5 5.5A.5.5 0 0 1 6 6v6a.5.5 0 0 1-1 0V6a.5.5 0 0 1 .5-.5Zm2.5 0a.5.5 0 0 1 .5.5v6a.5.5 0 0 1-1 0V6a.5.5 0 0 1 .5-.5Zm3 .5a.5.5 0 0 0-1 0v6a.5.5 0 0 0 1 0V6Z"
                />
                <path
                  d="M14.5 3a1 1 0 0 1-1 1H13v9a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V4h-.5a1 1 0 0 1-1-1V2a1 1 0 0 1 1-1H6a1 1 0 0 1 1-1h2a1 1 0 0 1 1 1h3.5a1 1 0 0 1 1 1v1ZM4.118 4 4 4.059V13a1 1 0 0 0 1 1h6a1 1 0 0 0 1-1V4.059L11.882 4H4.118ZM2.5 3h11V2h-11v1Z"
                />
              </svg>
            </button>
            <div
              className="invalid-feedback"
            >
              The maximum size allowed is 1MB.
            </div>
          </div>
        </div>
        <div>
          <p>
            (*) means mandatory field
          </p>
        </div>
        <div
          className="text-center"
        >
          <button
            className="btn btn-primary"
            type="submit"
          >
            Submit
          </button>
        </div>
      </form>
    </div>
  </div>
</div>
`;
